packages/core/src/__tests__/TextInput.test.js

```diff
@@ -21,6 +26,51 @@ describe('<TextInputBasicRow> helper component', () => {
     });
 });
 
+describe('<InnerInput> helper component', () => {
+    it('renders an <input type="text"> by default, with inputProps', () => {
+        const wrapper = shallow(<InnerInput inputProps={{ foo: 'bar' }} />);
+
+        expect(wrapper.matchesElement(
+            <input type="text" foo="bar" />
+        )).toBeTruthy();
+    });
+
+    it('renders an <AutoSizeTextarea> with defaults in multi-line mode, with inputProps', () => {
+        const wrapper = shallow(<InnerInput multiLine />);
+
+        expect(wrapper.matchesElement(
+            <AutoSizeTextarea minRows={2} />
+        )).toBeTruthy();
+
+        wrapper.setProps({
+            minRows: 5,
+            maxRows: 9,
+        });
+        expect(wrapper.matchesElement(
+            <AutoSizeTextarea minRows={5} maxRows={9} />
+        )).toBeTruthy();
+
+        wrapper.setProps({
```

Comment:
Maybe this (L53 - L58) could be combined together with the test code above (L45 - L51)